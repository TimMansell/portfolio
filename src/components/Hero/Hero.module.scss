.hero {
  height: 100vh;
}

.img {
  position: fixed;
  top: 0;
  width: 100vw;
  height: 100vh;

  &::before {
    content: '';
    position: fixed;
    top: 0;
    width: 100vw;
    height: 100vh;
    background: rgba(0, 0, 0, 0.05);
  }
}

.transition {
  &::before {
    animation: transition 2s ease;
  }
}

@keyframes transition {
  50% {
    background: var(--colour-1);
  }
}
