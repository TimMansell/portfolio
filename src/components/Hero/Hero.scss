@import 'utilities/breakpoints';

.hero {
  height: 100vh;

  &__img {
    @include media-breakpoint-up(md) {
      background-image: url('./img/hero.jpg');
      background-position: center;
      background-size: cover;
      position: fixed;
      top: 0;
      width: 100%;
      height: 100%;
      filter: blur(3px);
    }
  }
}
