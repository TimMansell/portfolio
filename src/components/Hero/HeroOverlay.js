import React from 'react';

import useScrollBlur from 'hooks/useScrollBlur';

import styles from './Hero.module.scss';

export const HeroOverlay = () => {
  const scrollStyles = useScrollBlur(0, 10);

  return <div className={styles.overlay} style={scrollStyles}></div>;
};

export default HeroOverlay;
