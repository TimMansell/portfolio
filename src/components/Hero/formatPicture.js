export const formatPicture = ({ name, title }) => {
  const src = `Hero/img/${name}`;

  const srcSizes = [
    {
      media: '(orientation: portrait)',
      sizes: ['640', '768', '1280', '1536', '2048'],
    },
    {
      media: '(orientation: landscape)',
      sizes: ['1366', '1600', '1920', '2560'],
    },
  ];

  const types = ['jpg'];

  return {
    title,
    src,
    types,
    srcSizes,
    isFullscreen: true,
  };
};
