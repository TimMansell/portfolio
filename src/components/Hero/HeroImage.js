import React, { useState, useEffect } from 'react';

import Picture from '../Picture';
import { formatPicture } from './formatPicture';
import styles from './Hero.module.scss';

import images from './json/images.json';

const TIMER = 7000;

export const Hero = () => {
  const [heroImages, setHeroImages] = useState(images);

  const [primaryImage, preloadImage] = heroImages;

  const primaryPicture = formatPicture(primaryImage);
  const preloadPicture = formatPicture(preloadImage);

  useEffect(() => {
    const timer = setTimeout(() => {
      setHeroImages(([firstImage, ...remainingImages]) => [
        ...remainingImages,
        firstImage,
      ]);
    }, TIMER);
    return () => clearTimeout(timer);
  }, [heroImages]);

  return (
    <div className={styles.hero}>
      <div className={styles.img} data-e2e="hero-img" data-timer={TIMER}>
        <Picture {...primaryPicture} />
        <Picture {...preloadPicture} aria-hidden="true" />
      </div>
    </div>
  );
};

export default Hero;
