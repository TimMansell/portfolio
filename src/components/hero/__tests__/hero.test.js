import React from 'react';
import { shallow } from 'enzyme';

import Hero from '../hero';

describe("Hero", () => {
  it("should render my component", () => {
    const props = {
      blurFrom: 0,
      blurTo: 1
    };

    const wrapper = shallow(<Hero {...props}/>);
  });
});
