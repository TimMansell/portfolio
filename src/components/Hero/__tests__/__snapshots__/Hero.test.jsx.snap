// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Hero > should match snapshot 1`] = `
<div
  className="_hero_77f2c2"
>
  <div
    className="_img_77f2c2"
    data-e2e="hero-img"
    data-timer={7000}
    style={
      {
        "filter": "blur(0px)",
      }
    }
  >
    <picture
      data-e2e="picture"
      title="Various gaming consoles and peripherals"
    >
      <source
        data-test="picture-source-0"
        media="(orientation: portrait)"
        srcSet="file:///H:/projects/portfolio/src/components/Hero/img/webp/lorenzo-herrera-p0j-mE6mGo4-unsplash-640.webp 640w, file:///H:/projects/portfolio/src/components/Hero/img/webp/lorenzo-herrera-p0j-mE6mGo4-unsplash-768.webp 768w, file:///H:/projects/portfolio/src/components/Hero/img/webp/lorenzo-herrera-p0j-mE6mGo4-unsplash-1280.webp 1280w, file:///H:/projects/portfolio/src/components/Hero/img/webp/lorenzo-herrera-p0j-mE6mGo4-unsplash-1536.webp 1536w, file:///H:/projects/portfolio/src/components/Hero/img/webp/lorenzo-herrera-p0j-mE6mGo4-unsplash-2048.webp 2048w"
        type="image/webp"
      />
      <source
        data-test="picture-source-1"
        media="(orientation: landscape)"
        srcSet="file:///H:/projects/portfolio/src/components/Hero/img/webp/lorenzo-herrera-p0j-mE6mGo4-unsplash-1366.webp 1366w, file:///H:/projects/portfolio/src/components/Hero/img/webp/lorenzo-herrera-p0j-mE6mGo4-unsplash-1600.webp 1600w, file:///H:/projects/portfolio/src/components/Hero/img/webp/lorenzo-herrera-p0j-mE6mGo4-unsplash-1920.webp 1920w, file:///H:/projects/portfolio/src/components/Hero/img/webp/lorenzo-herrera-p0j-mE6mGo4-unsplash-2560.webp 2560w"
        type="image/webp"
      />
      <source
        data-test="picture-source-2"
        media="(orientation: portrait)"
        srcSet="file:///H:/projects/portfolio/src/components/Hero/img/jpg/lorenzo-herrera-p0j-mE6mGo4-unsplash-640.jpg 640w, file:///H:/projects/portfolio/src/components/Hero/img/jpg/lorenzo-herrera-p0j-mE6mGo4-unsplash-768.jpg 768w, file:///H:/projects/portfolio/src/components/Hero/img/jpg/lorenzo-herrera-p0j-mE6mGo4-unsplash-1280.jpg 1280w, file:///H:/projects/portfolio/src/components/Hero/img/jpg/lorenzo-herrera-p0j-mE6mGo4-unsplash-1536.jpg 1536w, file:///H:/projects/portfolio/src/components/Hero/img/jpg/lorenzo-herrera-p0j-mE6mGo4-unsplash-2048.jpg 2048w"
        type="image/jpg"
      />
      <source
        data-test="picture-source-3"
        media="(orientation: landscape)"
        srcSet="file:///H:/projects/portfolio/src/components/Hero/img/jpg/lorenzo-herrera-p0j-mE6mGo4-unsplash-1366.jpg 1366w, file:///H:/projects/portfolio/src/components/Hero/img/jpg/lorenzo-herrera-p0j-mE6mGo4-unsplash-1600.jpg 1600w, file:///H:/projects/portfolio/src/components/Hero/img/jpg/lorenzo-herrera-p0j-mE6mGo4-unsplash-1920.jpg 1920w, file:///H:/projects/portfolio/src/components/Hero/img/jpg/lorenzo-herrera-p0j-mE6mGo4-unsplash-2560.jpg 2560w"
        type="image/jpg"
      />
      <img
        alt="Various gaming consoles and peripherals"
        className="_img_ec3fd8 _fullscreen_ec3fd8"
        data-e2e="picture-img"
        data-test="picture-img"
        loading="auto"
        src="file:///H:/projects/portfolio/src/components/Hero/img/jpg/lorenzo-herrera-p0j-mE6mGo4-unsplash-2560.jpg"
      />
    </picture>
    <picture
      data-e2e="picture"
      title="Snow-capped trees with aurora borealis in background"
    >
      <source
        data-test="picture-source-0"
        media="(orientation: portrait)"
        srcSet="file:///H:/projects/portfolio/src/components/Hero/img/webp/maria-vojtovicova-SPvJPDXEmqA-unsplash-640.webp 640w, file:///H:/projects/portfolio/src/components/Hero/img/webp/maria-vojtovicova-SPvJPDXEmqA-unsplash-768.webp 768w, file:///H:/projects/portfolio/src/components/Hero/img/webp/maria-vojtovicova-SPvJPDXEmqA-unsplash-1280.webp 1280w, file:///H:/projects/portfolio/src/components/Hero/img/webp/maria-vojtovicova-SPvJPDXEmqA-unsplash-1536.webp 1536w, file:///H:/projects/portfolio/src/components/Hero/img/webp/maria-vojtovicova-SPvJPDXEmqA-unsplash-2048.webp 2048w"
        type="image/webp"
      />
      <source
        data-test="picture-source-1"
        media="(orientation: landscape)"
        srcSet="file:///H:/projects/portfolio/src/components/Hero/img/webp/maria-vojtovicova-SPvJPDXEmqA-unsplash-1366.webp 1366w, file:///H:/projects/portfolio/src/components/Hero/img/webp/maria-vojtovicova-SPvJPDXEmqA-unsplash-1600.webp 1600w, file:///H:/projects/portfolio/src/components/Hero/img/webp/maria-vojtovicova-SPvJPDXEmqA-unsplash-1920.webp 1920w, file:///H:/projects/portfolio/src/components/Hero/img/webp/maria-vojtovicova-SPvJPDXEmqA-unsplash-2560.webp 2560w"
        type="image/webp"
      />
      <source
        data-test="picture-source-2"
        media="(orientation: portrait)"
        srcSet="file:///H:/projects/portfolio/src/components/Hero/img/jpg/maria-vojtovicova-SPvJPDXEmqA-unsplash-640.jpg 640w, file:///H:/projects/portfolio/src/components/Hero/img/jpg/maria-vojtovicova-SPvJPDXEmqA-unsplash-768.jpg 768w, file:///H:/projects/portfolio/src/components/Hero/img/jpg/maria-vojtovicova-SPvJPDXEmqA-unsplash-1280.jpg 1280w, file:///H:/projects/portfolio/src/components/Hero/img/jpg/maria-vojtovicova-SPvJPDXEmqA-unsplash-1536.jpg 1536w, file:///H:/projects/portfolio/src/components/Hero/img/jpg/maria-vojtovicova-SPvJPDXEmqA-unsplash-2048.jpg 2048w"
        type="image/jpg"
      />
      <source
        data-test="picture-source-3"
        media="(orientation: landscape)"
        srcSet="file:///H:/projects/portfolio/src/components/Hero/img/jpg/maria-vojtovicova-SPvJPDXEmqA-unsplash-1366.jpg 1366w, file:///H:/projects/portfolio/src/components/Hero/img/jpg/maria-vojtovicova-SPvJPDXEmqA-unsplash-1600.jpg 1600w, file:///H:/projects/portfolio/src/components/Hero/img/jpg/maria-vojtovicova-SPvJPDXEmqA-unsplash-1920.jpg 1920w, file:///H:/projects/portfolio/src/components/Hero/img/jpg/maria-vojtovicova-SPvJPDXEmqA-unsplash-2560.jpg 2560w"
        type="image/jpg"
      />
      <img
        alt="Snow-capped trees with aurora borealis in background"
        className="_img_ec3fd8 _fullscreen_ec3fd8"
        data-e2e="picture-img"
        data-test="picture-img"
        loading="auto"
        onLoad={[Function]}
        src="file:///H:/projects/portfolio/src/components/Hero/img/jpg/maria-vojtovicova-SPvJPDXEmqA-unsplash-2560.jpg"
      />
    </picture>
  </div>
</div>
`;
